src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -46,7 +46,7 @@ public IDirectoryContents GetDirectoryContents(string subpath)
             }
             else
             {
-                return InnerProvider.GetDirectoryContents(physicalPath);
+                return InnerProvider.GetDirectoryContents(physicalPath.Value);
```

Comment:
This was being implicitly converted to "string" and that would encode spaces as "%20". The fix simply passes down the raw value (which was already decoded by hosting) to the underlying provider